examples/jupyter/tensor_addition.ipynb

```diff
@@ -0,0 +1,226 @@
+    "ADDRESS_SPACE_C = sym.ADDRESS_SPACE_C"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d6f8cd53",
+   "metadata": {},
+   "source": [
+    "#### (2) Constraints\n",
+    "To specify how we want to distribute the different dimensions of our problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "16ef23d5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define constraints for the kernel\n",
+    "constraints = [\n",
+    "    # specifies how computation is tiled\n",
+    "    tkw.WorkgroupConstraint(M, BLOCK_M, 0),\n",
+    "    tkw.WorkgroupConstraint(N, BLOCK_N, 1),\n",
+    "    tkw.WaveConstraint(M, BLOCK_M / 2),\n",
+    "    tkw.WaveConstraint(N, BLOCK_N / 2),\n",
+    "    tkw.HardwareConstraint(\n",
+    "        threads_per_wave=64,\n",
+    "        vector_shapes={M: BLOCK_M, N: BLOCK_N}\n",
+    "    )\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "399e2fd2",
+   "metadata": {},
+   "source": [
+    "#### (3) Kernel Definition\n",
+    "Our tensor addition example with compute C = A + B. They are 2D matrices of dimension MxN in f16 precision."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3db5ed64",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The kernel\n",
+    "@tkw.wave(constraints)\n",
+    "def matrix_add(\n",
+    "    # defines matrix in memory of req dimension with specific data types\n",
+    "    a: Memory[M, N, ADDRESS_SPACE_A, f16],\n",
+    "    b: Memory[M, N, ADDRESS_SPACE_B, f16],\n",
+    "    c: Memory[M, N, ADDRESS_SPACE_C, f16],\n",
+    "):\n",
+    "    # Intialize the accumulator register with zeroes\n",
+    "    c_reg = Register[M,N,f16](0.0)\n",
+    "\n",
+    "    # loads values from memory into registers\n",
+    "    a_reg = tkw.read(a)\n",
+    "    b_reg = tkw.read(b)\n",
+    "\n",
+    "    # compute the sum\n",
+    "    c_reg = a_reg + b_reg\n",
+    "    \n",
+    "    # writing results back to memory\n",
+    "    tkw.write(c_reg, c)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89e6e72e",
+   "metadata": {},
+   "source": [
+    "#### (4) Testing the kernel \n",
+    "A function to verify our implementation works by comparing with PyTorch reference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "97168291",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The kernel testing function\n",
+    "def test_gemm():\n",
+    "    # Create test matrices\n",
+    "    m, n = 128, 128\n",
+    "\n",
+    "    # Initialize input matrices with random values\n",
+    "    torch.manual_seed(0)\n",
+    "    a = torch.randn(m, n, dtype=torch.float16, device=\"cuda\")\n",
+    "    b = torch.zeros(m, n, dtype=torch.float16, device=\"cuda\")\n",
+    "    c = torch.zeros(m, n, dtype=torch.float16, device=\"cuda\")\n",
+    "\n",
+    "    # Set hyperparameters for compilation\n",
+    "    hyperparams = {\n",
```

Comment:
Lets come up with a better name for this and potentially wrap it to make it look cleaner.